tests/ContainerFactoryTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Joindin\Api\Test;
+
+use Joindin\Api\ContainerFactory;
+use PHPUnit\Framework\TestCase;
+use Psr\Container\ContainerInterface;
+
+class ContainerFactoryTest extends TestCase
+{
+    private $config = [
+        'akismet' => [
+            'apiKey' => 'key',
+            'blog'   => 'blog'
+        ],
+        'email' => [
+            'contact' => 'excample@example.com',
+            "from" => "example@example.com",
+            'smtp' => [
+                'host'     => 'localhost',
+                'port'     => 25,
+                'username' => 'username',
+                'password' => 'ChangeMeSeymourChangeMe',
+                'security' => null,
+            ],
+        ],
+        'website_url' => 'www.example.com'
+    ];
+
+    /**
+     * @test
+     */
+    public function containerIsCreated()
+    {
+        $this->assertInstanceOf(ContainerInterface::class, ContainerFactory::build($this->config));
+    }
+
+    /**
+     * @test
+     * @covers \Joindin\Api\ContainerFactory::build
+     *
+     * @dataProvider dataProvider
+     * @param string $service
+     */
+    public function serviceIsDefined($service)
+    {
+        $container = ContainerFactory::build($this->config, true);
+        $this->assertTrue($container->has($service));
+    }
+
+    /**
+     * @test
+     * @covers \Joindin\Api\ContainerFactory::build
+     *
+     * @dataProvider dataProvider
+     * @param string $service
+     */
+    public function servicesCanBeCreated($service)
+    {
+        $container = ContainerFactory::build($this->config, true);
+        $this->assertInstanceOf($service, $container->get($service));
+    }
+
+    /**
+     * @test
+     * @covers \Joindin\Api\ContainerFactory::build
+     */
+    public function spamCheckServiceIsNullCheckerWhenDisabled()
+    {
+        $container = ContainerFactory::build([], true);
+        $this->assertTrue($container->has(\Joindin\Api\Service\SpamCheckServiceInterface::class));
```

Comment:
Now that we're using namespaces, the `\` prefix means global namespace, so I believe this should just be 

```php
$this->assertTrue($container->has(Joindin\Api\Service\SpamCheckServiceInterface::class));
```

Better yet would be to add a `use Joindin\Api\Service\SpanCheckServiceInterface` line to the top of this file and just have this line be 

```php
$this->assertTrue($container->has(SpamCheckServiceInterface::class));
```

Reply:
Agreed.